### PR TITLE
feat(dark-mode): Add option to toggle Settings UI theme

### PR DESCRIPTION
**Pull request by CoderOO7**
## Changes
* Implement dark mode for changing Settings Menu theme.

tabliss_darkMode

This feature is implemented for #320 pull request. The theme is not limited to settings menu only, but can be used in other components too. If you have any query regarding code then please ping me or if any changes need to done.

### CODE REVIEW

**Comment by nazibalalam** [comment]
Awesome!

**Comment by CoderOO7** [comment]
In previous commit their was a small ui bug i.e initially toggle button color was black, fix that in this commit.
If anyone need it can take latest pull. And eager to know where I can improve the code. Thanks :)

File: src/views/settings/Settings.sass [hunk 1]
```diff
@@ -1,21 +1,35 @@
 .Settings
+    &[data-theme="dark"]
+        --background-color: #2f2f2f
+        --text-color: white
```
Comment (joelshepherd) [inline src/views/settings/Settings.sass:4]: So we change this to a media query that uses `prefers-color-scheme: dark` instead.

**Review by joelshepherd — CHANGES_REQUESTED** [review]
Looks quite slick!

I prefer not needing to manually set the theme of every app on my system. Can we use `prefers-color-scheme: dark` instead of a manual control: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

You might also want to set `color-scheme` to change native UI elements too (like dropdowns): https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme

Thanks!

File: src/views/settings/Settings.sass (hunk 1 above)
Comment (CoderOO7) [reply src/views/settings/Settings.sass]: I have done changes that you asked. Let me know if their is something else where I can do improvement in code. Thanks for the suggestion & review.

In new pull request I have done following changes:
1. add prefers-color-scheme media query for dark mode
2. refactor the previous code
3. Used @mixin to keep code dry.
4. Move the main `styles.css` in seperate folder `styles`, where add some new configs for sass

Thanks :)

**Review by nazibalalam — COMMENTED** [review]
seems great man! Not sure why you selected me to review it...

**Comment by mscwilson** [comment]
This is very cool! I found this PR looking to see if it's possible to set the Tabliss background image to change over when the system mode changes. My background is a plain colour, I'd like to be able to choose a light mode colour and a dark mode colour, so they can switch over automatically. Is that possible with your feature here?

File: src/views/settings/Settings.sass (hunk 1 above)
Comment (joelshepherd) [reply src/views/settings/Settings.sass]: Sorry for taking a while to get back to you.

Yeah manual control is fine too, but the default should be "auto" which respects the OS scheme. So there would be 3 settings for theme: `light`, `dark`, and `auto`.

I am happy to make the adjustments if you like (especially given you've probably moved on from the PR a while ago :joy:).

**Comment by awesmubarak** [comment]
As  I understand this PR has resolved the requested change as of df2cff4 ? So would it be possible to merge it?

**Comment by joelshepherd** [comment]
The current release about to go out focused on under-the-hood changes with data storage. The release after I will be focusing on a revamp on the settings UI, and will return to this.

**Comment by ericpa06** [comment]
This might be interesting for some users. It's enables dark mode on this addon, just add this code on custom css. I only tested on Firefox, so I'm not sure if will work on Chromium browsers :(

```
@media (prefers-color-scheme: dark) {

.Settings input[type="email"].primary, .Settings input[type="file"].primary, .Settings input[type="number"].primary, .Settings input[type="text"].primary, .Settings input[type="time"].primary, .Settings input[type="url"].primary, .Settings textarea.primary, .Settings select.primary, .Settings input[type="email"], .Settings input[type="file"], .Settings input[type="number"], .Settings input[type="text"], .Settings input[type="time"], .Settings input[type="url"], .Settings textarea, .Settings select {
  border: 2px solid #ecf0f1; !important;
}



#root div.Settings div.plane {
background-color: #1e1e1e!important
}

.Settings .Widget {
  background: #2b2a33;
}

.Settings .Widget .button--icon {
  background-color: #3082b8;
  color: white;
}

label {
color: white;
}

textarea, div label select {
background: black !important;
color: white!important;
}

Settings .plane {
  background-color: #353535 !important;
  color: green;
}

.Settings h2, .Settings h3, .Settings h4 {
  color: #3498db;
}

div.Widget:nth-child(6) {
display: none
}

}
```

**Comment by Waradu** [comment]
i've made my own custom dark mode i think it looks kinda nice:
```css
.Overlay svg {
    opacity: 0;
}
.Overlay:hover svg {
    color: #ffffff;
    opacity: 1;
}
.Slot.middleCentre {
    width: 420px;
}
.Settings .plane {
    background-color: #1e1e1e;
    color: #b3b4b6;
}
.Settings .Widget {
    background: #262626;
}
.Settings input[type=email].primary, .Settings input[type=file].primary, .Settings input[type=number].primary, .Settings input[type=text].primary, .Settings input[type=time].primary, .Settings input[type=url].primary, .Settings textarea.primary, .Settings select.primary, .Settings select {
    background-color: #262626;
    border: 0.25em solid #262626;
}
h1, h2, h3, h4, h5 {
    color: white !important;
}
textarea, input {
    background: #262626;
    color: #c3c4c6!important;
    outline: 0;
}
```

Bildschirmfoto 2022-12-05 um 14 24 48

**Pull request closed without merging**